### PR TITLE
Update deps required to build with go1.24-27093581

### LINKED ISSUE #813: sync_diff_inspector: doesn't build with gotip
## Bug Report

1. What did you do?

With
```
go version devel go1.24-27093581 Sat Aug 17 03:03:34 2024 +0000 linux/amd64
```

```
make GO=gotip sync_diff_inspector
```

2. What did you expect to see?

Successful build

4. What did you see instead?

```
dvaneeden@dve-carbon:~/dev/pingcap/tidb-tools$ make GO=gotip sync_diff_inspector 
gotip build -ldflags '-X "github.com/pingcap/tidb-tools/pkg/utils.Version=v7.5.3-2-gd226440" -X "github.com/pingcap/tidb-tools/pkg/utils.BuildTS=2024-08-19 08:17:42" -X "github.com/pingcap/tidb-tools/pkg/utils.GitHash=d226440121147098eb5eb99cbc1efb94092ec68e" -X "github.com/pingcap/tidb-tools/pkg/utils.GitBranch=master"' -o bin/sync_diff_inspector ./sync_diff_inspector
# github.com/pingcap/tidb-tools/sync_diff_inspector
link: github.com/pingcap/tidb/pkg/util/printer: invalid reference to runtime.buildVersion
make: *** [Makefile:60: sync_diff_inspector] Error 1
```

5. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

6. which tool are you using?

sync-diff-inspector

7. what version of tool are you using (`pump -V` or `tidb-lightning -V` or `syncer -V`)?

## Related

- https://github.com/pingcap/tidb/issues/53618

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #813 ref #819

Also resolves https://github.com/pingcap/tidb-tools/security/dependabot/28

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

**Comment by wuhuizuo** [comment]
See the beautiful fix in tidb.
The fix is ​​more elegant. The main thing here is to update the version of the tidb dependency. This change is forward compatible with the golang version: the runtime.Version() method exists at least in go 1.0.0.

File: go.mod [hunk 1]
```diff
@@ -5,49 +5,50 @@ go 1.21
...
 	github.com/siddontang/go v0.0.0-20180604090527-bdc77568d726
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.9.0
-	github.com/tikv/pd/client v0.0.0-20240322051414-fb9e2d561b6e
+	github.com/tikv/pd/client v0.0.0-20240805092608-838ee7983b78
 	go.etcd.io/etcd/client/v3 v3.5.12
 	go.etcd.io/etcd/tests/v3 v3.5.12
 	go.uber.org/atomic v1.11.0
 	go.uber.org/zap v1.27.0
-	google.golang.org/grpc v1.62.1
```
Comment (wuhuizuo) [inline go.mod]: should we upgrade it in another PR, also for `github.com/BurntSushi/toml`.

**Comment by dveeden** [comment]
```
dvaneeden@dve-carbon:~/dev/pingcap/tidb-tools$ govulncheck ./...
=== Symbol Results ===

Vulnerability #1: GO-2024-2918
    Azure Identity Libraries Elevation of Privilege Vulnerability in
    github.com/Azure/azure-sdk-for-go/sdk/azidentity
  More info: https://pkg.go.dev/vuln/GO-2024-2918
  Module: github.com/Azure/azure-sdk-for-go/sdk/azidentity
    Found in: github.com/Azure/azure-sdk-for-go/sdk/azidentity@v1.5.1
    Fixed in: github.com/Azure/azure-sdk-for-go/sdk/azidentity@v1.6.0
    Example traces found:
      #1: pkg/utils/util.go:50:30: utils.GetJSON calls ioutil.ReadAll, which eventually calls azidentity.ClientSecretCredential.GetToken

Your code is affected by 1 vulnerability from 1 module.
This scan also found 0 vulnerabilities in packages you import and 1
vulnerability in modules you require, but your code doesn't appear to call these
vulnerabilities.
Use '-show verbose' for more details.
```

File: go.mod (hunk 1 above)
Comment (dveeden) [reply go.mod]: So for grpc and toml you want separate PR's?

Comment (wuhuizuo) [reply go.mod]: Yes, I recommend keeping it single-purpose

> So for grpc and toml you want separate PR's?

**Review by Defined2014 — APPROVED** [review]

**Review by mjonss — APPROVED** [review]

**Pull request merged**